drivers/hmis_external_apis/extensions/hmis/hud/client_extension.rb

```diff
@@ -18,6 +18,16 @@ module ClientExtension
                   -> { where(namespace: HmisExternalApis::AcHmis::Mci::SYSTEM_ID) },
                   class_name: 'HmisExternalApis::ExternalId',
                   as: :source
+
+          # Used by ClientSearch concern
+          def self.search_by_external_id(where, text)
+            eid_t = HmisExternalApis::ExternalId.arel_table
+            matches_external_value = eid_t[:source_type].eq(sti_name).and(eid_t[:value].eq(text))
+            client_ids = HmisExternalApis::ExternalId.where(matches_external_value).pluck(:source_id)
+            return where unless client_ids.any?
+
+            where.or(c_t[:id].in(client_ids))
```

Comment:
Oh, I missed this, not sure that's how we should handle this...  Sorry, looking at `text_searcher` I think we should treat this more like the `numeric` section and just return the client_ids that match.  Put them into the list of client_ids we'll then call where on at the end.  (Am I missing something?)  Otherwise we're going to end up with something like where(id: ids).or(where(id: other_ids)) and depending on what we're doing with the resulting scope, it could get unhappy.

Reply:
Yeah it will end up with something like `c_t[:id].in([1,2,3]).or(c_t[:id].eq(4))`. Cant postgres handle that? We have other searches that will yield multiple "or"s on the same field, like checking if FirstName matches the string OR if FirstName is in the homophone list.

text_searcher yields that `where` block and then we call `scope.where(where).pluck(:id)`.

Reply:
Postgres won't have an issue, but ActiveRecord gets really unhappy if the two or segments don't have compatible joins....  Probably not a problem, but since we're already plucking source ids, it seems like we could just tack them on the existing array and have one `where id in (1, 2, 3...)` instead.

Reply:
Oh ok, I see. Like this? https://github.com/greenriver/hmis-warehouse/pull/3084/commits/668fd7fb8a910d929ca5c9e10feadf76058edc22

Reply:
Actually no, that won't work, because it bypasses the "viewable by" scope I think. Anything that comes after `client_ids = block.call(where)` is not taking the search scope into account. So I think doing it this way would return clients that you don't have permission to see.